rest/context.go

```diff
@@ -155,6 +152,11 @@ type gorillaRequestContext struct {
 	messages []string
 }
 
+func setValueOnRequestContext(req *http.Request, key, val interface{}) *http.Request {
+	ctx := context.WithValue(req.Context(), key, val)
+	return req.WithContext(ctx)
+}
+
 // NewContext returns a RequestContext populated with parameters from the request path and
 // query string.
 func NewContext(parent context.Context, req *http.Request, writer http.ResponseWriter) RequestContext {
```

Comment:
This, along with just about any other reference to `context.Context` is gonna be a breaking change technically. Should probably do a major version bump